node/pkg/governor/governor_prices.go

```diff
@@ -38,31 +40,32 @@ func (gov *ChainGovernor) initCoinGecko(ctx context.Context) error {
 	params.Add("vs_currencies", "usd")
 
 	if first {
-		if gov.logger != nil {
-			gov.logger.Info("cgov: did not find any securities, nothing to do!")
-		}
-
+		gov.logger.Info("cgov: did not find any tokens, nothing to do!")
 		return nil
 	}
 
 	gov.coinGeckoQuery = "https://api.coingecko.com/api/v3/simple/price?" + params.Encode()
+	gov.logger.Info("cgov: coingecko query: ", zap.String("query", gov.coinGeckoQuery))
 
-	if gov.logger != nil {
-		gov.logger.Info("cgov: coingecko query: ", zap.String("query", gov.coinGeckoQuery))
+	if err := supervisor.Run(ctx, "govpricer", gov.PriceQuery); err != nil {
+		return err
 	}
 
+	return nil
+}
+
+func (gov *ChainGovernor) PriceQuery(ctx context.Context) error {
 	timer := time.NewTimer(time.Millisecond) // Start immediately.
-	go func() {
-		for {
-			select {
-			case <-ctx.Done():
-				return
-			case <-timer.C:
-				gov.queryCoinGecko()
-				timer = time.NewTimer(time.Duration(5) * time.Minute)
-			}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-timer.C:
+			gov.queryCoinGecko()
+			timer = time.NewTimer(time.Duration(5) * time.Minute)
 		}
-	}()
+	}
 
 	return nil
```

Comment:
The linter has a point here, it is impossible here to exit the loop, so no error will be returned.